CONTRIBUTING.md

```diff
@@ -0,0 +1,193 @@
+        -   [Clone your fork](#clone-a-repo).
+        -   [Do your work on a branch in your fork](#do-your-work-on-a-branch-in-your-fork).
+    -   1.B: You have made changes.
+        -   1.B.1: You are working on the main branch.
+            -   1.B.1.A: You made commits.
+                -   TODO
+            -   1.B.1.B: You haven't made commits.
+                -   [Stash your changes](#stash-your-changes).
+                -   [Create a branch](#create-a-branch).
+                -   [Create a fork](#create-a-fork).
+                -   [Add the remote of your fork](#add-a-remote).
+                -   [Push the branch to your fork](#push-a-branch).
+                -   [Clone your fork](#clone-a-repo).
+                -   [Do your work on a branch in your fork](#do-your-work-on-a-branch-in-your-fork).
+        -   1.B.2: You are working on a different branch.
+            -   1.B.2.A: You have made commits.
+                -   [Create a fork](#create-a-fork).
+                -   [Add the remote of your fork](#add-a-remote).
+                -   [Push the branch to your fork](#push-a-branch).
+                -   [Clone your fork](#clone-a-repo).
+                -   [Do your work on a branch in your fork](#do-your-work-on-a-branch-in-your-fork).
+            -   1.B.2.B: You haven't made commits.
+                -   [Commit your changes](#commit-changes).
+                -   [Create a fork](#create-a-fork).
+                -   [Add the remote of your fork](#add-a-remote).
+                -   [Push the branch to your fork](#push-a-branch).
+                -   [Clone your fork](#clone-a-repo).
+                -   [Do your work on a branch in your fork](#do-your-work-on-a-branch-in-your-fork).
+-   2: You have a fork of a Numberscope repository.
+    -   2.A: You haven't made changes.
+        -   [Clone your fork](#clone-a-repo) (if you don't have it on your
+            computer yet).
+        -   [Create a branch](#create-a-branch).
+        -   [Do your work on a branch in your fork](#do-your-work-on-a-branch-in-your-fork).
+    -   2.B: You have made changes.
+        -   2.B.1: You are working on the main branch.
+            -   2.B.1.A: You made commits.
+                -   TODO
+            -   2.B.1.B: You haven't made commits.
+                -   [Stash your changes](#stash-your-changes).
+                -   [Create a branch](#create-a-branch).
+                -   [Do your work on a branch in your fork](#do-your-work-on-a-branch-in-your-fork).
+        -   2.B.2: You are working on a different branch.
+            -   2.B.2.A: You have made commits.
+                -   [Do your work on a branch in your fork](#do-your-work-on-a-branch-in-your-fork).
+            -   2.B.2.B: You haven't made commits.
+                -   [Commit your changes](#commit-changes).
+                -   [Do your work on a branch in your fork](#do-your-work-on-a-branch-in-your-fork).
+
+## Stash your changes
+
```

Comment:
All these sections here (one for each command) are so clean and clear and crisp, it's very easy to navigate.  I like this style.